From 5c4025b4a6283927eba5f8a3e69b17b9c46b6a4b Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Sun, 13 Aug 2017 11:36:15 +0200
Subject: [PATCH 1/4] gdk-pixbuf-xlib: Mark public funcs _GDK_PIXBUF_EXTERN

Without this, the meson-built libgdk_pixbuf_xlib has no exports at all.

https://bugzilla.gnome.org/show_bug.cgi?id=786232
---
 contrib/gdk-pixbuf-xlib/gdk-pixbuf-xlib.h    |  7 +++++++
 contrib/gdk-pixbuf-xlib/gdk-pixbuf-xlibrgb.h | 24 +++++++++++++++++++++++-
 2 files changed, 30 insertions(+), 1 deletion(-)

diff --git a/contrib/gdk-pixbuf-xlib/gdk-pixbuf-xlib.h b/contrib/gdk-pixbuf-xlib/gdk-pixbuf-xlib.h
index c75554681..d81db1b36 100644
--- a/contrib/gdk-pixbuf-xlib/gdk-pixbuf-xlib.h
+++ b/contrib/gdk-pixbuf-xlib/gdk-pixbuf-xlib.h
@@ -30,8 +30,10 @@ G_BEGIN_DECLS
 
 /* init */
 
+_GDK_PIXBUF_EXTERN
 void gdk_pixbuf_xlib_init (Display *display, int screen_num);
 
+_GDK_PIXBUF_EXTERN
 void gdk_pixbuf_xlib_init_with_depth (Display *display, int screen_num,
 				      int prefDepth);
 
@@ -39,12 +41,14 @@ void gdk_pixbuf_xlib_init_with_depth (Display *display, int screen_num,
 
 /* render */
 
+_GDK_PIXBUF_EXTERN
 void gdk_pixbuf_xlib_render_threshold_alpha (GdkPixbuf *pixbuf, Pixmap bitmap,
 					     int src_x, int src_y,
 					     int dest_x, int dest_y,
 					     int width, int height,
 					     int alpha_threshold);
 
+_GDK_PIXBUF_EXTERN
 void gdk_pixbuf_xlib_render_to_drawable (GdkPixbuf *pixbuf,
 					 Drawable drawable, GC gc,
 					 int src_x, int src_y,
@@ -54,6 +58,7 @@ void gdk_pixbuf_xlib_render_to_drawable (GdkPixbuf *pixbuf,
 					 int x_dither, int y_dither);
 
 
+_GDK_PIXBUF_EXTERN
 void gdk_pixbuf_xlib_render_to_drawable_alpha (GdkPixbuf *pixbuf,
 					       Drawable drawable,
 					       int src_x, int src_y,
@@ -64,6 +69,7 @@ void gdk_pixbuf_xlib_render_to_drawable_alpha (GdkPixbuf *pixbuf,
 					       XlibRgbDither dither,
 					       int x_dither, int y_dither);
 
+_GDK_PIXBUF_EXTERN
 void gdk_pixbuf_xlib_render_pixmap_and_mask (GdkPixbuf *pixbuf,
 					     Pixmap *pixmap_return,
 					     Pixmap *mask_return,
@@ -73,6 +79,7 @@ void gdk_pixbuf_xlib_render_pixmap_and_mask (GdkPixbuf *pixbuf,
 
 /* drawable */
 
+_GDK_PIXBUF_EXTERN
 GdkPixbuf *gdk_pixbuf_xlib_get_from_drawable (GdkPixbuf *dest,
 					      Drawable src,
 					      Colormap cmap, Visual *visual,
diff --git a/contrib/gdk-pixbuf-xlib/gdk-pixbuf-xlibrgb.h b/contrib/gdk-pixbuf-xlib/gdk-pixbuf-xlibrgb.h
index f838ea807..689d4f75f 100644
--- a/contrib/gdk-pixbuf-xlib/gdk-pixbuf-xlibrgb.h
+++ b/contrib/gdk-pixbuf-xlib/gdk-pixbuf-xlibrgb.h
@@ -51,7 +51,7 @@
 #ifndef __XLIB_RGB_H__
 #define __XLIB_RGB_H__
 
-#include <glib.h>
+#include <gdk-pixbuf/gdk-pixbuf.h>
 
 G_BEGIN_DECLS
 
@@ -71,17 +71,22 @@ struct _XlibRgbCmap {
   unsigned char lut[256]; /* for 8-bit modes */
 };
 
+_GDK_PIXBUF_EXTERN
 void
 xlib_rgb_init (Display *display, Screen *screen);
+_GDK_PIXBUF_EXTERN
 void
 xlib_rgb_init_with_depth (Display *display, Screen *screen, int prefDepth);
 
+_GDK_PIXBUF_EXTERN
 unsigned long
 xlib_rgb_xpixel_from_rgb (guint32 rgb);
 
+_GDK_PIXBUF_EXTERN
 void
 xlib_rgb_gc_set_foreground (GC gc, guint32 rgb);
 
+_GDK_PIXBUF_EXTERN
 void
 xlib_rgb_gc_set_background (GC gc, guint32 rgb);
 
@@ -109,6 +114,7 @@ typedef enum
   XLIB_RGB_DITHER_MAX
 } XlibRgbDither;
 
+_GDK_PIXBUF_EXTERN
 void
 xlib_draw_rgb_image (Drawable drawable,
 		     GC gc,
@@ -120,6 +126,7 @@ xlib_draw_rgb_image (Drawable drawable,
 		     unsigned char *rgb_buf,
 		     int rowstride);
 
+_GDK_PIXBUF_EXTERN
 void
 xlib_draw_rgb_image_dithalign (Drawable drawable,
 			       GC gc,
@@ -133,6 +140,7 @@ xlib_draw_rgb_image_dithalign (Drawable drawable,
 			       int xdith,
 			       int ydith);
 
+_GDK_PIXBUF_EXTERN
 void
 xlib_draw_rgb_32_image (Drawable drawable,
 			GC gc,
@@ -144,6 +152,7 @@ xlib_draw_rgb_32_image (Drawable drawable,
 			unsigned char *buf,
 			int rowstride);
 
+_GDK_PIXBUF_EXTERN
 void
 xlib_draw_gray_image (Drawable drawable,
 		      GC gc,
@@ -155,12 +164,15 @@ xlib_draw_gray_image (Drawable drawable,
 		      unsigned char *buf,
 		      int rowstride);
 
+_GDK_PIXBUF_EXTERN
 XlibRgbCmap *
 xlib_rgb_cmap_new (guint32 *colors, int n_colors);
 
+_GDK_PIXBUF_EXTERN
 void
 xlib_rgb_cmap_free (XlibRgbCmap *cmap);
 
+_GDK_PIXBUF_EXTERN
 void
 xlib_draw_indexed_image (Drawable drawable,
                          GC gc,
@@ -175,34 +187,44 @@ xlib_draw_indexed_image (Drawable drawable,
 
 /* Below are some functions which are primarily useful for debugging
    and experimentation. */
+_GDK_PIXBUF_EXTERN
 Bool
 xlib_rgb_ditherable (void);
 
+_GDK_PIXBUF_EXTERN
 void
 xlib_rgb_set_verbose (Bool verbose);
 
 /* experimental colormap stuff */
+_GDK_PIXBUF_EXTERN
 void
 xlib_rgb_set_install (Bool install);
 
+_GDK_PIXBUF_EXTERN
 void
 xlib_rgb_set_min_colors (int min_colors);
 
+_GDK_PIXBUF_EXTERN
 Colormap
 xlib_rgb_get_cmap (void);
 
+_GDK_PIXBUF_EXTERN
 Visual *
 xlib_rgb_get_visual (void);
 
+_GDK_PIXBUF_EXTERN
 XVisualInfo *
 xlib_rgb_get_visual_info (void);
 
+_GDK_PIXBUF_EXTERN
 int
 xlib_rgb_get_depth (void);
 
+_GDK_PIXBUF_EXTERN
 Display *
 xlib_rgb_get_display (void);
 
+_GDK_PIXBUF_EXTERN
 Screen *
 xlib_rgb_get_screen (void);
 
-- 
2.14.1

